packages/angular_devkit/build_angular/src/browser/tests/options/watch_spec.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { mergeMap, take, tap, timeout } from 'rxjs/operators';
+import { buildWebpackBrowser } from '../../index';
+import { BASE_OPTIONS, BROWSER_BUILDER_INFO, describeBuilder } from '../setup';
+
+describeBuilder(buildWebpackBrowser, BROWSER_BUILDER_INFO, (harness) => {
+  describe('Option: "watch"', () => {
+    it('does not wait for file changes when false', (done) => {
+      harness.useTarget('build', {
+        ...BASE_OPTIONS,
+        watch: false,
+      });
+
+      // If the build waits then it will timeout with a custom 30 second timeout
+      let count = 0;
+      harness
+        .execute()
+        .pipe(timeout(30000))
+        .subscribe({
+          complete() {
+            expect(count).toBe(1);
+            done();
+          },
+          next({ result }) {
+            count++;
+            expect(result?.success).toBe(true);
+          },
+          error(error) {
+            done.fail(error);
+          },
+        });
+    });
+
+    it('watches for changes to the main file', async () => {
+      harness.useTarget('build', {
+        ...BASE_OPTIONS,
+        main: 'src/main.ts',
+        watch: true,
+      });
+
+      let count = 0;
+      await harness
+        .execute()
+        .pipe(
+          timeout(30000),
+          tap(({ result }) => {
+            count++;
+            expect(result?.success).toBe(true);
+            if (count === 1) {
+              harness.expectFile('dist/main.js').content.not.toContain('abcd1234');
+            } else if (count === 2) {
+              harness.expectFile('dist/main.js').content.toContain('abcd1234');
+            }
+          }),
+          mergeMap(async () => {
+            if (count === 1) {
+              return harness.modifyFile(
+                'src/main.ts',
+                (content) => content + 'console.log("abcd1234");',
+              );
+            }
+          }),
```

Comment:
Maybe the below is simpler?

```suggestion
        concatMap(async (_, index) => {
          switch (index) {
            case 0:
              harness.expectFile('dist/main.js').content.not.toContain('abcd1234');
              await harness.modifyFile('src/main.ts', (content) => content + 'console.log("abcd1234");'),
              break;
            case 1:
              harness.expectFile('dist/main.js').content.toContain('abcd1234');
              break;
          }
        }),
```

Reply:
Yes, sounds good.